documentation/devonfw-ide-contribution-getting-started.asciidoc

```diff
@@ -0,0 +1,71 @@
+
+=== Commandlet
+Use an existing commandlet like https://github.com/devonfw/ide/blob/master/scripts/src/main/resources/scripts/command/helm[HELM] as a reference. Create a file with the name of the tool and put it in the folder https://github.com/devonfw/ide/tree/master/scripts/src/main/resources/scripts/command[scripts/command]. The first part of the script contains the autocompletion, which should return the available parameters. Functions like setup or run or other additional functions follow. Finally, it defines which function should be executed depending on the parameter.
+
+Use the `doInstall` method for the installation, which downloads the tool based on the mirrors, unpacks it if necessary and puts it in the software folder in the Devon installation `doInstall "-" "[installation folder]" "[tool name]" "[version]"`.
+
+=== LICENSE
+Find the license for the new tool and check if the license exists in the https://github.com/devonfw/ide/blob/master/documentation/LICENSE.asciidoc[document]. If the license does not exist, add it to the file and complete the table of contents.
+
+== Settings
+The https://github.com/devonfw/ide-settings[IDE settings] allow project-specific configurations of the IDE tools. For your own settings, fork the repository, adjust the configuration and specify the repository URL during installation. Important notes on configuration and especially configuration files can be found https://devonfw.com/website/pages/docs/devonfw-guide_ide.wiki_configuration.asciidoc.html[here]. In the configuration files many variables can be defined, listed https://github.com/devonfw/ide/blob/master/documentation/variables.asciidoc[here].
+
+== Tipps and tricks
+
+=== Bash (Linux's Borne Again Shell)
+**For beginners:**
+
+You probably know this black box on Windows (the Command Prompt) that accepts commands like 'dir' that shows the content of the directory you are currently in,
+or 'cd' with which you can change the directory you are currently in and many more commands that allow you to interact with your Windows Operating System in a more direct and eventually faster way.  
+Now for users of a Linux Distribution there is something similar, but way more powerfull, usually called "Bourne Again Shell", or in short Bash. This Command Line Interface or "Shell" is what we use most often when implementing something new for our Command-Line-Tool with the name devon-IDE.  https://www.youtube.com/watch?v=I4EWvMFj37g[Have a look at Bash in 150 Seconds.]
+For MacOS you often also use BASH. That's why it is often enough to implement code for Mac or Linux (at least to some extend as you'll see soon enough) to be able to use it on both Operating Systems.
+
+Let's get to the practical part of the Bash-usage. 
+
+**If you are using Windows**, _then please install WSL so you will be able to use Bash as well. E.g. https://www.youtube.com/watch?v=qYlgUDKKK5A[watch this] guide untill Minute 5:58 for a basic setup on Windows and some command examples._
```

Comment:
I do not get this.
Simply use `git-bash` and you have a full bash for windows without the need for WSL.

Reply:
I made a compromise with a commit a few minutes ago if you are ok with that, by mentioning `git-bash` as first choice, as we primarily use it during our development. As addition I mentioned `WSL` as it can be of value. For example, when using doIsLinux `WSL` will return a 0, whereas `git-bash` won't, what can be usefull while testing. Still, maybe this argument is to weak to keep mentioning WSL?